shared/schemas/workflow-status-schema.json

```diff
@@ -0,0 +1,83 @@
+{
+    "$schema": "http://json-schema.org/draft-07/schema#",
+    "title": "WorkflowStatus",
+    "description": "Status of workflow execution",
+    "type": "object",
+    "properties": {
+        "status": {
```

Comment:
What about properties like `startedAt`, `finishedAt`?

Reply:
good idea

Reply:
those values are being fetched from the data-index that stores the information about the running instances.
the workflow itself shouldn't report that information.
Generally speaking, workflow result in this context is about the application output that the user should know about, in respect to the input parameters.

Meaning, if the user provided input `X`, the workflow result should say (in case of a detected problem with `X`) if the result is a failure. This will provide an extra information, e.g error message, or an output related to the workflow context (e.g. what is the name and link of the created resources)

However, the `startedAt` and `finishedAt` will not be provided from the workflow, since it isn't the responsibility of the workflow developer to assign those values to the output structure.

In addition, there is a `status` field for the workflow run, in which case it determines how the workflow execution was completed, and based on those values, the UI offers actions to the users, such as abort, rerun or retrigger.

However, `Completed` status will be reported even for a workflow that ends with a failure and for that purpose we'd like to include the `status` in this context (at least until we'll get a new feature from sonataflow implemented with adding an indication for an error. cc @gabriel-farache - what this feature released already?)

Reply:
The workflow's developer can enhance https://github.com/apache/incubator-kie-kogito-runtimes/issues/3475 and add to the end state an `errorMessage` property to the metadata. By setting a value for that `errorMessage` the workflow will end with `Error` state.

However, I wouldn't like the workflow developer to report the actual expected status of a workflow, and rely on the values from the DataIndex's processInstances (aka workflow runs) for that.

See example: https://github.com/apache/incubator-kie-kogito-examples/pull/1912/files#diff-f73c639cb03eff5222587f103f21fcf8d886291ccb12c6c23ed69cd72a628188R13

Reply:
See https://github.com/parodos-dev/serverless-workflows/pull/135 to see how the workflow developer can set the `errorMessage` value. Not sure if setting this value also set the status to error, I'll check

Reply:
I confirm that is the `errorMessage` is set, the status of the workflow is set to `error` as well

Reply:
Thanks. That means we do not need the `completedWith` field. If the workflow ends with error, the `errorMessage` will be used to provide human-friendly description.

In any case, the UI will combine both the instance's error message and the result's message.

Reply:
@mareklibra @gabriel-farache showed us yesterday that although the status is set to error when setting the errorMessage, the metadata.errorMessage was not propogated to the instance the UI gets
So I think we still need to confirm this

Reply:
I think @ydayagi has to confirm that (https://issues.redhat.com/browse/FLPATH-1674)

Reply:
Let's wait for confirmation then, I got confused with the comments in this thread.

If I read correctly, the errorMessage is meant to do that. So if not, it's a bug and we should evaluate the complexity of fixing it versus using `completedWith`.